app/src/androidTest/java/com/github/sdpcoachme/messaging/InAppNotifierTest.kt

```diff
@@ -145,6 +148,75 @@ class InAppNotifierTest {
         }
     }
 
+    @Test
+    fun onMessageReceivedEnablesEmailRecoveryAndOpensChatActivityWhenUserReceivesNotificationThenLoggsOutAndThenClicksOnIt() {
```

Comment:
Regarding the long test name, it would be good to have a small comment that describes what the test is for and maybe why it is necessary.